pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -714,15 +725,17 @@ func conditionallyConvertToIstioMtls(
 	serviceAccounts []string,
 	sni string,
 	proxy *model.Proxy,
-	serviceMTLSMode authn_model.MutualTLSMode,
-) *networking.TLSSettings {
+	autoMTLSEnabled bool,
+	meshExternal bool,
+) (*networking.TLSSettings, mtlsContextType) {
+	mtlsCtx := userSupplied
 	if tls == nil {
-		if serviceMTLSMode != authn_model.MTLSStrict {
-			// Destination service is not in strict-mTLS mode, do nothing.
-			return nil
+		if meshExternal || !autoMTLSEnabled {
```

Comment:
This mean even authn policy is STRICT, we still need DR if `autoMTLSEnabled==false`. It's fine by me, but just want to make sure it is intentional. You'll need to update the `global-mtls-on-no-dr.yaml` in the reachability test suite.  + @rshriram

Reply:
Yes. It will be the old 1.3 behavior